crates/metrics/src/listener/events/sync.rs

```diff
@@ -0,0 +1,64 @@
+use crate::listener::MetricsListener;
+use metrics::Gauge;
+use reth_metrics_derive::Metrics;
+use reth_primitives::{
+    stage::{StageCheckpoint, StageId},
+    BlockNumber,
+};
+use std::collections::HashMap;
+
+/// Pipeline and live sync metrics collection.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum SyncMetricEvent {
+    /// Stage reached new checkpoint.
+    StageCheckpoint {
+        /// Stage ID.
+        stage_id: StageId,
+        /// Stage checkpoint.
+        checkpoint: StageCheckpoint,
+        /// Maximum known block number reachable by this stage.
+        /// If specified, `entities_total` metric is updated.
+        max_block_number: Option<BlockNumber>,
+    },
+}
```

Comment:
I'm 100% convinced this belongs in the metrics crate.

I think of the metrics crate as somethings that provides support for metrics only.
imo this belongs somewhere else, perhaps stage-sync

Reply:
fair, moved to `stages`. Live sync will use it later to send stage checkpoint updates.